File: src3/Notification.java
public class Notification {
    private String message;   // Notification message
    private String date;      // Notification date (e.g., "YYYY-MM-DD")
    private boolean isRead;   // Read status of the notification

    // Constructor
    public Notification(String message, String date) {
        this.message = message != null ? message : "No message";  // Ensure non-null message
        this.date = date != null ? date : "Unknown date";          // Ensure non-null date
        this.isRead = false;                                      // Default is unread
    }

    // Getter and Setter methods
    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        if (message != null && !message.trim().isEmpty()) {
            this.message = message;
        } else {
            System.out.println("Invalid message. Cannot be empty or null.");
        }
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        if (date != null && !date.trim().isEmpty()) {
            this.date = date;
        } else {
            System.out.println("Invalid date. Cannot be empty or null.");
        }
    }

    public boolean isRead() {
        return isRead;
    }

    // Mark the notification as read
    public void markAsRead() {
        this.isRead = true;
    }

    // Mark the notification as unread
    public void markAsUnread() {
        this.isRead = false;
    }

    // Display notification details
    public void displayNotification() {
        System.out.println("Message: " + message);
        System.out.println("Date: " + date);
        System.out.println("Status: " + (isRead ? "Read" : "Unread"));
    }
}

File: Iteration3/Python/Notification.py
class Notification:


    def __init__(self, message, date):

        self.message = message
        self.date = date
        self.is_read_status = False  # Default is unread

    def get_message(self):

        return self.message

    def set_message(self, message):

        self.message = message

    def get_date(self):

        return self.date

    def set_date(self, date):

        self.date = date

    def is_read(self):

        return self.is_read_status

    def mark_as_read(self):

        self.is_read_status = True

    def set_read(self, is_read):

        self.is_read_status = is_read

    def __str__(self):

        status = "Read" if self.is_read_status else "Unread"
        return f"[{status}] {self.date}: {self.message}"
